packages/bugsnag_flutter/lib/src/client.dart

```diff
@@ -234,6 +237,50 @@ abstract class BugsnagClient {
 
   /// Removes a previously added "on error" callback.
   void removeOnError(BugsnagOnErrorCallback onError);
+
+
+  networkInstrumentation(data) {
+    if (data is! Map<String, dynamic>) return true;
+    String status = data["status"];
+    String requestId = data["request_id"];
+    if (status == "started") {
+      _onRequestStarted(requestId);
+    } else if (status == "complete") {
+      _onRequestComplete(requestId, data);
+    }
+    return true;
+  }
+
+  void _onRequestStarted(String requestId) {
+    final stopwatch = Stopwatch()..start();
+    _openNetworkRequests[requestId] = stopwatch;
+  }
+
+  void _onRequestComplete(String requestId, data) {
+    final stopwatch = _openNetworkRequests.remove(requestId);
+    if (stopwatch != null) {
+      final duration = stopwatch.elapsedMilliseconds;
+      final String clientName = data["client"];
+      String params = "";
+      if(data["url"].split("?").length > 1){
+        params = data["url"].split("?").last;
+      }
+      final int statusCode = data["status_code"];
+      final String status = statusCode < 400 ? "succeeded" : "failed";
+      leaveBreadcrumb("$clientName request $status", metadata: {
+        "duration": duration,
+        "method": data["http_method"],
+        "url": data["url"].split("?").first,
+        if(params.isNotEmpty) "urlParams": params,
+        if(data["request_content_length"] != null && data["request_content_length"] > 0) "requestContentLength": data["request_content_length"],
+        if(data["response_content_length"] != null && data["response_content_length"] > 0) "responseContentLength": data["response_content_length"],
+        if(data["status_code"] != null) "status": data["status_code"],
+      }, type: BugsnagBreadcrumbType.request);
```

Comment:
It feels like this could be a bit more efficient by only splitting the `data["url"]` once.

The whole method also feels like it should be wrapped in a `try {} catch {}` just in case something unexpected comes along (like `data["url"]` not being a `String`).